src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -159,6 +166,29 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                     }
                 }
 
+                if(!await AreCentralVersionRequirementsSatisfiedAsync(cpvmEnabled))
+                {
+                    await MSBuildRestoreUtility.ReplayWarningsAndErrorsAsync(_request.ExistingLockFile?.LogMessages, _logger);
```

Comment:
I think the warnings/errors raised in here will not end up in the assets file. Like wise they don't end up in the cache file. 
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs#L325-L333

So maybe there's some issues clearing them? Idk really know the consequences but I just can see that it's inconsistent. Note that this is a problem with the current lock file implemtnation as well, not just this change. I will create a future task for that.

Reply:
Issue https://github.com/NuGet/Home/issues/9173

Reply:
Discussed offline., Removed the Reply